lib/AsyncStorage.js

```diff
@@ -14,7 +14,9 @@
 const {NativeModules} = require('react-native');
 
 const RCTAsyncStorage =
-  NativeModules.RNC_AsyncSQLiteDBStorage || NativeModules.RNCAsyncStorage;
+  NativeModules.RNC_AsyncSQLiteDBStorage ||
+  NativeModules.RNCAsyncStorage ||
+  NativeModules.AsyncLocalStorage; // NativeModules.AsyncLocalStorage provides compatibility with out out tree platform
```

Comment:
Could we move this to the top? That would allow custom implementation on "in-tree" platforms.

Reply:
I don't quite get what "tree platform" is. I don't see a problem with moving it on top.